debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -319,8 +321,13 @@ else if (!connectorConfig.getSnapshotMode().shouldStream()) {
 
     private void addSchemaEvent(RelationalSnapshotContext<MySqlPartition, MySqlOffsetContext> snapshotContext,
                                 String database, String ddl) {
-        schemaEvents.addAll(databaseSchema.parseSnapshotDdl(snapshotContext.partition, ddl, database,
-                snapshotContext.offset, clock.currentTimeAsInstant()));
+
+        List<SchemaChangeEvent> schemaChangeEvents = databaseSchema.parseSnapshotDdl(snapshotContext.partition, ddl, database,
+                snapshotContext.offset, clock.currentTimeAsInstant());
+        List<SchemaChangeEvent> missingSchemaChangeEvents = schemaChangeEvents.stream()
+                .filter(not(schemaEvents::contains))
+                .collect(Collectors.toList());
```

Comment:
@mfvitale Hi here have a performance bottleneck.  time complexity is `O(databaseSchemaNumber²)`, We have 140+ databases on single instance, here  process need 2 hour+.

Reply:
Hi @minchowang can you please open a separate issue? Thanks.

Reply:
https://issues.redhat.com/browse/DBZ-7608